docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+
+When the `loadChassis` method is called with a file path as its argument, it first gets a `ClassLoader` instance from the `DataStorage` class and uses it to create an `InputStream` for the specified file. A `BufferedReader` is then created to read the contents of the file.
+
+The JSON file is read and concatenated into a single string using `reduce`. The GSON library is then used to deserialize the JSON string into a `Map<String, Chassis>`, with the key being the name of the chassis and the value being a `Chassis` object.
+
+Next, the method iterates over the `chassisMap` and sets the name attribute of each `Chassis` object to its corresponding key in the map.
+
+Finally, the method iterates over the `chassisMap` again, adding each `Chassis` object to the `ChassisList` stored in the `stringToComponentListMap` HashMap.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+
+![load Chassis Sequential Diagram](uml-pictures/loadChassisSequential.png)
+
+### Parser class
+The UML class diagram shows the features of the Parser class:
+
+![Parser Class Diagram](uml-pictures/ParserClass.png)
+
+The purpose of the Parser class is to parse the user input into commands for execution. With reference to the diagram,
+the Parser class does not have any attributes and only contains public methods for parsing of the user input.
+The methods in the Parser class creates a subclass which inherits from the abstract superclass Command class. 
+From the diagram, Command class has a private attribute named arguments which stores the user inputs as a String. The Command class
+contains methods that help to set and retrieve the private attribute in the class. 
+Overall, this class diagram provides an overview of the purpose of the Parser class.
+
+####  ViewerAddCommand() method
+This Sequential analysis will show how the ViewerAddCommand() method works, this will also serve as an example for the ViewerDeleteCommand and ViewerViewCommand.
+During the ParseCommand, when a add command is detected in the user input, it will call the `ViewerAddCommand` method with 
+a name as its argument. The `ViewerAddCommand` then calls a method to retrieve the argument.The `ViewerAddCommand`
+checks if the argument is empty and throws an `PPException` if it is empty. A `PC` object is created with the argument as
+an input. The `add` method is called by accessing the `dataStorage` and `pcList` classes to add the `PC` object to the list.
+Finally, a message is returned to inform the user that the `ViewerAddCommand` method has been executed successfully.
+
+A UML sequence diagram showing the interactions between the different objects involved in handling this method can be
+found below:
+![Viewer Add Command Sequential Diagram](uml-pictures/ViewerAddCommand.png)
+### PC Class Architecture
+
+![PC Class Diagram](uml-pictures/PcDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/78406251/228763244-4b388d24-d6d4-43b8-8f66-865fcd5e9e49.png)
![image](https://user-images.githubusercontent.com/78406251/228763309-26ce1d57-4bde-4f53-ab26-766b945d5e8f.png)

In the DG, the `Component` class is said to be an abstract class but in the diagram, it isn't shown as an abstract class. If I didn't misunderstand the diagram, perhaps this is a mistake?